### PR TITLE
SE-306 annotates all cocoon tests with Lusid Features

### PR DESCRIPTION
**Pull request by slacki123**
# Pull Request Checklist

- [x] Read the [contributing guidelines](https://github.com/finbourne/lusid-python-tools/blob/master/docs/CONTRIBUTING.md)
- [x] Tests pass

# Description of the PR

Annotates all cocoon test features with lusid_feature decorators

### CODE REVIEW

File: tests/integration/cocoon/test_cocoon_failures.py [hunk 1]
```diff
@@ -365,7 +385,7 @@ def test_load_from_data_frame_failure(
             ]
         ]
     )
-    def test_load_from_data_frame_failure(
+    def test_load_from_data_frame_failure2(
```
Comment (rizwansaeed) [inline tests/integration/cocoon/test_cocoon_failures.py:388]: why was this renamed to `2`? can we have a more descriptive name

**Review by rizwansaeed — CHANGES_REQUESTED** [review]

File: tests/integration/cocoon/test_cocoon_failures.py (hunk 1 above)
Comment (slacki123) [reply tests/integration/cocoon/test_cocoon_failures.py:388]: @rizwansaeed Turns out that if two methods have the exact same names in the same class, then only one of them is returned by the inspect method. That's the reason for name change

**Review by rizwansaeed — APPROVED** [review]
